exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/WrappedAccessor.java

```diff
@@ -327,6 +328,19 @@ public Array getArray() throws SQLException {
     }
   }
 
+  @Override
+  public Struct getStruct() throws SQLException {
+    try {
+      return delegate.getStruct();
+    } catch (RuntimeException e) {
+      String message = e.getMessage();
+      if (message != null && message.startsWith("cannot convert to")) {
+        throw new SQLException(e.getMessage(), e);
+      }
+      throw e;
```

Comment:
Looks like the error handling can be refactored to a separate method and used everywhere.